devices/Button/ButtonBase.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Threading;
+
+namespace Iot.Device.Button
+{
+    /// <summary>
+    /// Base implementation of Button logic.
+    /// Hardware independent. Inherit for specific hardware handling.
+    /// </summary>
+    public class ButtonBase : IDisposable
+    {
+        internal const int DEFAULT_DOUBLE_PRESS_TICKS = 15000000;
+        internal const int DEFAULT_HOLDING_MS = 2000;
+
+        private bool _disposed = false;
+
+        private int _doublePressMs;
+        private int _holdingMs;
+
+        private ButtonHoldingState _holdingState = ButtonHoldingState.Completed;
+
+        private long _lastPress = DateTime.MinValue.Ticks;
+        private Timer _holdingTimer;
+
+        public delegate void ButtonPressedDelegate(object sender, EventArgs e);
+        public delegate void ButtonHoldingDelegate(object sender, ButtonHoldingEventArgs e);
```

Comment:
Missing the /// comments as well and in the next ones

Reply:
Thank you, should be resolved now